sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -15,35 +18,94 @@ internal sealed class AzureSearchDocumentsEventSource : EventSource
+        public void RetryQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(RetryQueueResizedEvent, queueSize);
+            WriteEvent(RetryQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has submitted batch of size {1}.")]
-        public void BatchSubmitted(string endPoint, int batchSize)
+        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "{0} at {1} has submitted batch of size {2}.")]
+        public void BatchSubmitted(string publisherType, string endPoint, int batchSize)
         {
-            WriteEvent(BatchSubmittedEvent, endPoint, batchSize);
+            WriteEvent(BatchSubmittedEvent, publisherType, endPoint, batchSize);
         }
 
-        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has updated the starting batch action count from {1} to {2}.")]
-        public void BatchActionCountUpdated(string endPoint, int oldBatchCount, int newBatchCount)
+        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Warning, Message = "{0} at {1} has updated the starting batch action count from {2} to {3}.")]
+        public void BatchActionCountUpdated(string publisherType, string endPoint, int oldBatchCount, int newBatchCount)
         {
-            WriteEvent(BatchActionCountUpdatedEvent, endPoint, oldBatchCount, newBatchCount);
+            WriteEvent(BatchActionCountUpdatedEvent, publisherType, endPoint, oldBatchCount, newBatchCount);
+        }
+
+        [Event(PublisherDisposedWithPendingActionsEvent, Level = EventLevel.Error, Message = "{0}: {1} unsent indexing actions at {2}.")]
+        public void PublisherDisposedWithPendingActions(string componentType, string endPoint, int indexingActionsCount)
+        {
+            WriteEvent(PublisherDisposedWithPendingActionsEvent, componentType, indexingActionsCount, endPoint);
+        }
+
+        [Event(DocumentsPublishedEvent, Level = EventLevel.Verbose, Message = "{0}: publishing documents. Flush = {2}")]
+        public void PublishingDocuments(string publisherType, bool flush)
+        {
+            WriteEvent(DocumentsPublishedEvent, publisherType, flush);
+        }
+
+        [Event(ActionNotificationEventHandlerExceptionThrownEvent, Level = EventLevel.Error, Message = "{0}: exception thrown at {1}. Action = {2}. Exception: {3}")]
+        public void ActionNotificationEventHandlerExceptionThrown(string componentType, string endPoint, string action, string e)
+        {
+            WriteEvent(ActionNotificationEventHandlerExceptionThrownEvent, componentType, endPoint, action, e);
+        }
+
+        [NonEvent]
+        public void ActionNotificationEventHandlerExceptionThrown(string componentType, string endPoint, string action, Exception e)
+        {
+            if (IsEnabled(EventLevel.Error, EventKeywords.All))
```

Comment:
What does this mean?  We only log these if errors are allowed?

Reply:
Yes. And the check is so we only do expensive formatting if the message is actually going to be logged.
I have removed the check because we have got rid of the formatting.